geomesa-hbase/geomesa-hbase-datastore/src/main/scala/org/locationtech/geomesa/hbase/coprocessor/KryoLazyDensityCoprocessor.scala

```diff
@@ -69,85 +69,115 @@ class KryoLazyDensityCoprocessor extends KryoLazyDensityService with Coprocessor
   def getService: Service = this
 
   def getDensity(controller: RpcController, request: KryoLazyDensityProto.DensityRequest, done: RpcCallback[KryoLazyDensityProto.DensityResponse]) {
-    val byteFilterArray = request.getByteFilter.toByteArray
     var response : DensityResponse = null
     var scanner : InternalScanner = null
+    var filterList : FilterList = new FilterList()
+
     try {
-      val scan = new Scan
-      val options: Map[String, String] = deserializeOptions(byteFilterArray)
+      val options: Map[String, String] = deserializeOptions(request.getOptions.toByteArray)
       val sft = SimpleFeatureTypes.createType("input", options(SFT_OPT))
+      var scanList : List[Scan] = List[Scan]()
+
+      options(RANGES_OPT).split(",").foreach(range => {
+        val scanInfo = range.split("\\|")
+        scanList ::= new Scan(Base64.decode(scanInfo(0)), Base64.decode(scanInfo(1)))
```

Comment:
would be better to use a mutable list implementation like ArrayBuffer for performance. since it's not exposed outside this function we don't have to worry about exposing mutable state.